keycache/keycache.go

```diff
@@ -159,11 +215,12 @@ func EncryptKey(in []byte, name string, override []byte) (out []byte, err error)
 // for RSA and EC keys, the cached RSA/EC key is used to decrypt
 // the pubEncryptedKey which is then used to decrypt the input
 // buffer.
-func DecryptKey(in []byte, name string, pubEncryptedKey []byte) (out []byte, err error) {
+func DecryptKey(in []byte, name, user string, labels []string, pubEncryptedKey []byte) (out []byte, err error) {
 	Refresh()
 
-	decryptKey, ok := matchUser(name)
+	decryptKey, ok := matchUser(name, user, labels)
 	if !ok {
+		log.Println("No Match", name, user, labels, UserKeys)
```

Comment:
Similar to the previous logging comment, this might be better as a note that no suitable decryptor could be found.